src/controllers/request.controller.js

```diff
@@ -45,15 +45,36 @@ export default class RequestController {
    * @returns {array} An array of objects or an empty one
    */
   static async findAll(req, res) {
-    // const requestData;
-
     await Request.findAll({ where: { user_id: req.body.user_id } })
-        .then((data) => {
-          if (data.length) {
-            return Response.Success(res, data, 200);
-          }
-          return Response.CustomError(res, 404, 'error', 'No requests found');
-        })
-        .catch((err) => Response.InternalServerError(res, err));
+      .then((data) => {
+        if (data.length) {
+          return Response.Success(res, data, 200);
+        }
+        return Response.CustomError(res, 404, 'error', 'No requests found');
+      })
+      .catch((err) => Response.InternalServerError(res, err));
+  }
+
+  /**
+     * @param {object} req The data which contains dates, reason, origin, destination etc
+     * @param {object} res Response returned to the user
+     * @returns {object} An object containing the request submitted into the db
+     */
+  static async TripRequests(req, res) {
+    const destination = [];
+    const {
+      user_id, category, origin, departure_date, reason, booking_id
+    } = req.body;
+    destination.push(req.body.destination);
+    const newRequest = {
+      user_id, category, origin, destination, departure_date, reason, booking_id
+    };
+
+    // persist booking to database
+    await Request.create(newRequest)
```

Comment:
Why are you using async/await and promise together here?. Why not stick to just async.